### ISSUE TITLE
It would be nice if developer can override openFileInput method

### ISSUE THREAD

**Issue by sfshine** [delight-im/Android-AdvancedWebView#210]
I have a feature that open a AlertDialog where  user can choose ''choose file from gallery" or "take picture directly"
I check the source code find that I cannot override the FileInput porcess logic that  AdvancedWebView defined.
It would be nice if developer can override openFileInput method

![image](https://user-images.githubusercontent.com/3046831/45069947-95bbda00-b101-11e8-88a9-7ab516cf0968.png)

**Comment by ocram** [comment]
You’re right. If we find the right abstraction, we should implement that. Thank you!

**Comment by anukools** [comment]
Any update on this as I am not getting "Choose from gallery or take picture" option from webview.

**Comment by ocram** [comment]
@anukools This issue here is just to simplify extending the functionality for file uploads in the future. If you want camera access *right now*, you should take a look at https://github.com/delight-im/Android-AdvancedWebView/issues/10, https://github.com/delight-im/Android-AdvancedWebView/issues/16, https://github.com/delight-im/Android-AdvancedWebView/issues/20 and https://github.com/delight-im/Android-AdvancedWebView/issues/81 instead.
